microsoft/testsuites/performance/networkperf.py

```diff
@@ -424,6 +435,568 @@ def perf_sockperf_latency_tcp_synthetic_busy_poll(self, result: TestResult) -> N
+        client.reboot()
+        server.reboot()
+
+        perf_tcp_pps(
+            test_result=result,
+            test_type="maxpps",
+            server=server,
+            client=client,
+            run_with_internal_address=True,
+        )
+
+    @TestCaseMetadata(
+        description="""
+        This test case uses ntttcp to test sriov tcp network throughput.
+        """,
+        priority=3,
+        timeout=TIMEOUT,
+        requirement=node_requirement(
+            node=schema.NodeSpace(
+                node_count=2,
+                memory_mb=search_space.IntRange(min=8192),
+            )
+        ),
+    )
+    def perf_tcp_ntttcp_passthrough_two_guest(self, result: TestResult) -> None:
+        environment = result.environment
+        assert environment, "fail to get environment from testresult"
+
+        client_node = cast(RemoteNode, environment.nodes[0])
+        client, client_nic_name = self._configure_passthrough_nic_for_node(client_node)
+        server_node = cast(RemoteNode, environment.nodes[1])
+        server, server_nic_name = self._configure_passthrough_nic_for_node(server_node)
+
+        # Reboot the nodes. Libvirt sometime re-use the nodes.
+        # Try to run the test on fresh state of the nodes
+        client.reboot()
+        server.reboot()
+
+        perf_ntttcp(
+            test_result=result,
+            client=client,
+            server=server,
+            server_nic_name=server_nic_name,
+            client_nic_name=client_nic_name,
+        )
+
+    @TestCaseMetadata(
+        description="""
+        This test case uses ntttcp to test sriov tcp network throughput.
+        """,
+        priority=3,
+        timeout=TIMEOUT,
+        requirement=node_requirement(
+            node=schema.NodeSpace(
+                node_count=2,
+                memory_mb=search_space.IntRange(min=8192),
+            )
+        ),
+    )
+    def perf_udp_1k_ntttcp_passthrough_two_guest(self, result: TestResult) -> None:
+        environment = result.environment
+        assert environment, "fail to get environment from testresult"
+
+        client_node = cast(RemoteNode, environment.nodes[0])
+        client, client_nic_name = self._configure_passthrough_nic_for_node(client_node)
+        server_node = cast(RemoteNode, environment.nodes[1])
+        server, server_nic_name = self._configure_passthrough_nic_for_node(server_node)
+
+        # Reboot the nodes. Libvirt sometime re-use the nodes.
+        # Try to run the test on fresh state of the nodes
+        client.reboot()
+        server.reboot()
+
+        perf_ntttcp(
+            test_result=result,
+            client=client,
+            server=server,
+            server_nic_name=server_nic_name,
+            client_nic_name=client_nic_name,
+            udp_mode=True,
+        )
+
+    def _configure_passthrough_nic_for_node(
+        self,
+        node: Node,
+    ) -> Tuple[RemoteNode, str]:
+        ctx = get_node_context(node)
```

Comment:
Avoid to use context in test cases, use features to operate nodes, so it's easy to support multiple platforms, and decouple cases and platforms.

Reply:
Sure, Will make this change